backend/src/resolvers/Mutation/GoalLink.js

```diff
@@ -0,0 +1,30 @@
+import { ForbiddenError } from 'apollo-server-core'
+import { checkAccess, Role, Privilege } from '../../util/accessControl'
+
+export const createGoalLink = async (root, {goalId, courseId, workspaceId}, context) => {
+  await checkAccess(context, {
+    minimumRole: Role.STAFF,
+    minimumPrivilege: Privilege.EDIT,
+    workspaceId
+  })
+  const course = nullShield(await context.prisma.course({ id: cousreId }))
```

Comment:
```suggestion
  const course = nullShield(await context.prisma.course({ id: courseId }))
```

(has this been tested?)